ui/app/AppLayouts/Profile/stores/ContactsStore.qml

```diff
@@ -16,6 +18,15 @@ QtObject {
         Component.onCompleted: {
             mainModuleInst.resolvedENS.connect(root.resolvedENS)
         }
+
+        readonly property var conn: Connections {
+            target: d.contactsModuleInst
+            function onTrustStatusRemoved(pubKey: string) {
+                const contactDetails = Utils.getContactDetailsAsJson(pubKey, false, false, true)
+                const displayName = SQUtils.Emoji.parse(ProfileUtils.displayName(contactDetails.localNickname, contactDetails.name, contactDetails.displayName, contactDetails.alias))
+                Global.displaySuccessToastMessage(qsTr("Trust mark removed for %1").arg(displayName))
+            }
+        }
```

Comment:
Yeah good ideas; I'll just probably emit a signal here, and catch it in the `ToastsManager`. One potential pitfall, the `ContactsStore` is not a toplevel one, it's created as a substore of `RootStore`

Option nr. 3 is something I had on mind too but seemed too complex to do here just for one action. The other (complexity) problem is that the proposed solution is simply incomplete: if you look at Popups.qml, we don't want to display a series of notifications as a result of performing single backend calls, but rather combine and group them in case we're performing several actions (cf Popups.qml, e.g. the noitifications inside `RemoveContactPopup`)

Reply:
One thing that just popped in my mind, is there a chance that we initialized more than one `ContactsStore`? If so, then we'd fire two toast messages?
That might be an argument to move the Connections to another place, though I don't have an idea where it should be :shrug:

Reply:
Yeah, if I understood correctly, it's another reason to decouple toast creation from the place where the action is called in UI. By observing model we can register events and emit notifications using `Qt.callLater` or even some delay, where the only the last call will be actually executed, aggregating all events into single nofification.

Reply:
@jrainville that's possible and it would lead to undesired behavior. But first thing is that store shouldn't do that to respect separation of concerns and the architecture we established via the guideline.

Reply:
So far it looks like the ContactsStore is really instantiated only once as part of the toplevel RootStore :) But one can never be too sure :D

Reply:
Done

In action:

![Snímek obrazovky z 2024-12-17 19-32-43](https://github.com/user-attachments/assets/0158f6ea-c038-4ae4-846c-97fa60a4a559)


![image](https://github.com/user-attachments/assets/dc522f79-69ea-4a98-b181-c14dd092276b)

Reply:
Got me thinking, would be a nice test for the stores if we don't instantiate them multiple times :laughing: Here it works, otherwise we'd get multiple notifications